### PR TITLE
finish dataTypes zh translate

### PR DESCRIPTION
**Pull request by lamberken**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

finish dataTypes zh translate.

### CODE REVIEW

**Comment by blinkov** [comment]
Looks like you haven't merged master to your branch for a while and your PR does not show diff from current translation...

cc @sundy-li @zhang2014

**Comment by lamber-ken** [comment]
I'm trying to solve the conflicts.

**Comment by blinkov** [comment]
@lamber-ken now PR shows that it has way too many unrelated changes

**Comment by sundy-li** [comment]
@blinkov  I have a question, should we keep the `[Original Artical](https://clickhouse.yandex/docs/en/data_types/boolean/) <!--hide-->`  links  in zh docs?

**Comment by blinkov** [comment]
> @blinkov I have a question, should we keep the `[Original Artical](https://clickhouse.yandex/docs/en/data_types/boolean/) <!--hide-->` links in zh docs?

@sundy-li It's supposed to be a link to where the rendered version of current article is published. So you can translate the words in `[...]` and replace `/en/` with `/zh/` in url, but if you don't it's no big deal.

File: docs/zh/data_types/array.md [hunk 1]
```diff
@@ -1,8 +1,84 @@
...
-# Array（T）
+# Array(T)
 
-一个包含类型 T 的 数组（Array）。T 类型可以是任意类型，包括数组类型。
-我们不推荐使用多维数组，因为多维数组并没有得到很好支持（比如，不能在 MergeTree 引擎的表中存储多维数组）。
+由 `T` 类型元素组成的数组。
 
-[来源文章](https://clickhouse.yandex/docs/en/data_types/array/) <!--hide-->
\ No newline at end of file
+`T` 可以是任意类型，包含数组类型。不推荐使用多维数组，ClickHouse 对多维数组的支持有限。例如，不能存储在 `MergeTree` 表中存储多维数组。
```
Comment (zhang2014) [inline docs/zh/data_types/array.md]: 不推荐使用多维数组 -> 但不推荐使用多维数组 is better.

File: docs/zh/data_types/array.md [hunk 2]
```diff
@@ -1,8 +1,84 @@
...
+
+## 创建数组
+
+您可以使用一个函数创建数组：
```
Comment (zhang2014) [inline docs/zh/data_types/array.md]: 您可以使用一个函数创建数组 -> 您可以使用array函数来创建数组 is better.

File: docs/zh/data_types/array.md [hunk 3]
```diff
@@ -1,8 +1,84 @@
...
+┌─x─────┬─toTypeName([1, 2])─┐
+│ [1,2] │ Array(UInt8)       │
+└───────┴────────────────────┘
+
+1 rows in set. Elapsed: 0.002 sec.
+```
+
+## 使用数据类型
+
+当动态创建数组时，ClickHouse 自动将参数类型定义为可以存储所有列出的参数的最窄的数据类型。如果存在任何 [NULL](../query_language/syntax.md#null-literal) 或者 [Nullable](nullable.md#data_type-nullable) 类型参数，那么数组元素的类型是 [Nullable](nullable.md#data_type-nullable)。
```
Comment (zhang2014) [inline docs/zh/data_types/array.md]: ClickHouse会自动根据数组中的元素计算出适合的类型。如果在元素中存在NULL或存在Nullable类型元素，那么数组的元素类型将会变成Nullable。

File: docs/zh/data_types/datetime.md [hunk 4]
```diff
@@ -1,16 +1,13 @@
...
-
-包含时间和日期。用4个字节来存储 Unix 时间戳（无符号的值）。允许存储的时间范围同 Date 类型。最小的值为 0000-00-00 00:00:00。
-DateTime 类型值精确到秒（不考虑闰秒）
+<a name="data_type-datetime"></a>
 
-## Time zones
+# DateTime
 
-在客户端或服务器启动时，会使用系统的时区来将时间日期从文本（划分成多个部分）到二进制之间相互转化。在文本格式中，有关夏令时信息不作存储。
+时间戳类型。用四个字节存储 Unix 时间戳（无符号的）。允许存储与日期类型相同的范围内的值。最小值为 0000-00-00 00:00:00。时间戳类型值精确到秒（没有闰秒）。
```
Comment (zhang2014) [inline docs/zh/data_types/datetime.md]: 用四个字节存储 Unix 时间戳（无符号的） -> 用四个字节（无符号的）存储 Unix 时间戳

**Comment by blinkov** [comment]
I'll apply comments above after merge.

**Comment by lamber-ken** [comment]
@sundy-li , @zhang2014, @blinkov  thanks for review and suggestion. :+1:

**Pull request merged**